1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,349 @@
+
+Non-goals:
+
+- Due to the relatively short length of a session compared to your PSReadline history,
+  PowerShell session history is not part of the historical suggestion
+- The extension framework doesn't allow a plugin implementation to call into the Runspace where
+  PSReadLine is running when providing predictive suggestions.
+- VSCode implementation. We agree this is important for a consistent experience and are working to
+  bring this feature in a future release, but the user experience for VS Code is outside the scope of this RFC.
+
+## Specification
+
+The proposal is to add Predictive IntelliSense and an extension model to improve the user
+interactive experience.
+
+### Key Design Considerations Predictive IntelliSense
+
+Predictive IntelliSense displays the most recent match command completion from command history. The
+prediction may change if a better match is found as the user types additional information. The
+predicted command may be accepted, edited, or ignored.
+
+![Prediction - Historical](./media/pred-wlc.png)
+
+### Install and Remove Predictive IntelliSense
+
+Predictive IntelliSense application programming interface (API) is implemented in the PowerShell
+engine and presented through the PSReadLine module. To receive the benefits of Predictive
+IntelliSense based on your PowerShell version, download and install the PSReadLine module from
+PSGallery
+
+The current release is PSReadLine 2.1.0 Beta 2:
+
+```powershell
+Install-Module PSReadLine -RequiredVersion 2.1.0-beta2 -AllowPrerelease
+```
+
+It is possible to unload or uninstall the PSReadLine module to disable Predictive IntelliSense. This
+creates a negative impact to customers as it would remove all features of PSReadLine. For this
+reason, Predictive IntelliSense has its own PSReadLine configuration to enable and disable
+predictions.
+
+Note - When upgrading PowerShell preview versions, it's important to upgrade to the corresponding
+version of PSReadLine. Incorrect versions will result in an error when PowerShell starts.
+
+### Enabling and Disabling Predictions
+
+By default, the feature Predictive IntelliSense is disabled. The impact of enabling this feature by
+default may cause new and existing users to become confused at the shell prompt. The feature is
+enabled and disabled with a PSReadLine command Set-PSReadLineOption that is executed at the shell
+prompt or in the user's Profile.
+
+Predictive IntelliSense in PSReadLine 2.1.0-beta2 currently supports the following arguments for
+prediction source;
+
+- None - This option disables Predictive IntelliSense
+- History - This option uses only the PSReadLine history for predictions
+
+Starting with PSReadLine 2.2.0-beta1, Predictive IntelliSense will support the following arguments
+for prediction source;
+
+- None - This option disables Predictive IntelliSense
+- History - This option uses only the PSReadLine history for predictions
+- HistoryAndPlugin - This option uses both for predictions
```

Comment:
Jason - change 142 to---History provides initial competitions, plugins provide additional plugins.

Reply:
Also - 142 -- change to "HistoryandPlugins"